app/script/controllers/add.product.controller.js

```diff
@@ -0,0 +1,19 @@
+(function () {
+    angular.module('app')
+        .controller('addProductController', function ($scope, apiService, $location) {
+            $scope.newProduct = function () {
+                apiService.addProduct($scope.product)
+                    .then(function () {
+                        $location.path('/products')
+                    });
+            };
+            (function() {
```

Comment:
I thought that it's necessery but now I see, it isn't. (It works without IIFE too.)